src/analyses/base.ml

```diff
@@ -1866,7 +1891,11 @@ struct
     let reachable = List.concat (List.map AD.to_var_may (reachable_vars (Analyses.ask_of_ctx ctx) (get_ptrs vals) ctx.global st)) in
     let reachable = List.filter (fun v -> CPA.mem v st.cpa) reachable in
     let new_cpa = CPA.add_list_fun reachable (fun v -> CPA.find v st.cpa) new_cpa in
-    {st' with cpa = new_cpa}
+    (* Identify locals of this fundec for which an outer copy (from a call down the callstack) is reachable *)
+    let reachable_other_copies = List.filter (fun v -> try (match Cilfacade.find_scope_fundec v with | Some fdec -> fdec.svar.vid = fundec.svar.vid |_ -> false) with _ -> false) reachable in
```

Comment:
This is quite difficult to read, maybe something like this instead:
```suggestion
    let is_outer v = match Cilfacade.find_scope_fundec v with
      | Some scope -> CilType.Fundec.equal scope fundec
      | None
      | exception Not_found -> false
    in
    let reachable_other_copies = List.filter is_outer reachable in
```

Although when does `find_scope_fundec` fail? It has an `option` return type, so it really shouldn't. Or if it does, then maybe a fix should be done in it.

Reply:
>  Although when does find_scope_fundec fail?

E.g. in the case of vars created by `malloc(...)` that were not in the globals hashtable when `varinfo_roles` was computed:

~~~C
void init (int *p, int x) {

}

int main () {
  int *p = malloc(sizeof(int));
  init(p,9);

  return 0;
}
~~~

Reply:
Ahh yes, of course. Then to be safe it indeed makes sense to change `find_scope_fundec` to return `None` also for those because other uses of that function could hit the same issue.